cluster-autoscaler/simulator/csi/provider/csinode_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	v1storagelister "k8s.io/client-go/listers/storage/v1"
+)
+
+// testCSINodeLister is a fake CSINode lister for testing.
+type testCSINodeLister struct {
+	csiNodes []*storagev1.CSINode
+}
+
+func (l *testCSINodeLister) List(selector labels.Selector) ([]*storagev1.CSINode, error) {
+	return l.csiNodes, nil
+}
+
+func (l *testCSINodeLister) Get(name string) (*storagev1.CSINode, error) {
+	for _, csiNode := range l.csiNodes {
+		if csiNode.Name == name {
+			return csiNode, nil
+		}
+	}
+	return nil, apierrors.NewNotFound(storagev1.Resource("csinode"), name)
+}
+
+func newTestCSINodeLister(csiNodes []*storagev1.CSINode) v1storagelister.CSINodeLister {
+	return &testCSINodeLister{csiNodes: csiNodes}
+}
+
+func TestNewCSINodeProvider(t *testing.T) {
```

Comment:
IMO this test should be converted to the table-driven format, and `TestSnapshot_EmptyLister` should be rolled in as one of the test cases. I know we still have a lot of these old-style imperative unit tests in CA, but they're really hard to follow and extend - we're trying to keep to the table-driven format for new ones (unless extremely small and simple).

Reply:
Moved it to table driven tests.